src/settings/LocationLocations/RemoteStorageDetails.js

```diff
@@ -15,7 +15,7 @@ const RemoteStorageDetails = ({ locationId }) => {
 
   const currentConfig = useMemo(() => {
     return configurations?.records.find(config => remoteMap[locationId] === config.id);
-  }, [locationId, remoteMap]);
+  }, [locationId, remoteMap, configurations]);
```

Comment:
Good catch! Does it solve the problem?
Honestly, I would get rid of `useMemo` here entirely